golang/src/main/java/com/regnosys/rosetta/generator/golang/enums/GolangEnumGenerator.xtend

```diff
@@ -0,0 +1,160 @@
+	}
+	
+	def Map<String, ? extends CharSequence>generate3(Iterable<RosettaEnumeration> rosettaEnums, String version) {
+		val result = new HashMap
+		val enums = rosettaEnums.sortBy[name].generateEnums3(version)
+		result.put(FILENAME,enums)
+		return result;
+		
+	}
+
+	def static toJavaEnumName(RosettaEnumeration enumeration, RosettaEnumValue rosettaEnumValue) {
+		return enumeration.name + '.' + EnumHelper.convertValues(rosettaEnumValue)
+	}
+
+	private def allEnumsValues(RosettaEnumeration enumeration) {
+		val enumValues = new ArrayList
+		var e = enumeration;
+
+		while (e !== null) {
+			e.enumValues.forEach[enumValues.add(it)]
+			e = e.superType
+		}
+		return enumValues.sortBy[name];
+	}
+
+	private def generateEnums(List<RosettaEnumeration> enums, String version)  '''		
+		package enums
+		
+		«fileComment(version)»			
+		
+		«FOR e : enums»
+			«val allEnumValues = allEnumsValues(e)»
+			«classComment(e.definition)»
+			type «e.name» int
+			const (
+			«FOR value: allEnumValues»
+				«methodComment(value.definition)»
+				«e.name»_«EnumHelper.convertValues(value)» «e.name» = iota + 1
+			«ENDFOR»
+			)
+		«ENDFOR»
+	'''
+	//generates struct representations of enums
+	private def generateEnums2(List<RosettaEnumeration> enums, String version){
+		val eString='''
+		package preenums
+		«FOR e : enums»					
+			type «e.name» int							
+		«ENDFOR»
+		'''
+		val enumDir = Files.createDirectories(Paths.get('''cdm/preenums'''))
+		Files.write(enumDir.resolve('''preenums.go'''), eString.toString.bytes)
+		
+		return	'''
+		package enums
+		import . "cdm/preenums"
+		«FOR e : enums»					
+		«fileComment(version)»			
+					
+			«val allEnumValues = allEnumsValues(e)»
+			«classComment(e.definition)»			
+			type «e.name.toLowerCase» struct {
+			«FOR value: allEnumValues»
+				«methodComment(value.definition)»
+				«EnumHelper.convertValues(value)» «e.name»
+			«ENDFOR»
+			}
+		«ENDFOR»
+		
+		«FOR e : enums»					
+			var «e.name.toUpperCase» = «e.name.toLowerCase»{}							
+		«ENDFOR»
+		
+		func init(){
+			«FOR e : enums»
+				«val allEnumValues = allEnumsValues(e)»
+				«var counter=1»				
+				«FOR value: allEnumValues»								
+					«e.name.toUpperCase».«EnumHelper.convertValues(value)» = «e.name»(«counter.toString»)
+					«{counter++; null}»
+				«ENDFOR»				
+			«ENDFOR»
+		}	
+		'''.replaceTabsWithSpaces
+		
+		
+	}
+	
+	private def generateEnums3(List<RosettaEnumeration> enums, String version){
+		for (e: enums){
+			val eString = '''		
+		«fileComment(version)»			
+			package «e.name»
+			import . "cdm/enums"
+			«val allEnumValues = allEnumsValues(e)»
+			«classComment(e.definition)»
+			
+			const (
+			«FOR value: allEnumValues»
+				«methodComment(value.definition)»
+				«EnumHelper.convertValues(value)» «e.name» = iota + 1
+			«ENDFOR»
+			)		
+		'''.replaceTabsWithSpaces
+		val enumDir = Files.createDirectories(Paths.get('''cdm/«e.name»'''))
+		Files.write(enumDir.resolve('''«e.name».go'''), eString.toString.bytes)
```

Comment:
No need to write the files. You only need to return the content of the enums as string, which gets added in `GolangCodeGenerator.java.afterGenerate()` method into a map.

Reply:
We had to resort to this side-effect because of a Golang's peculiarity. It does not really have enums -- they use named constants instead. Furthermore, Golang does not allow to have the same name for constants in the same package even if they have different types. There is a lot of these conflicts in our case. Because of these two reasons we have to create a bunch of packages and thus folders for the enums. So we cannot just pass a string, which is then to be written to a single file/package.

Reply:
If that's the case, then you can either:
1. generate (as oppose to write files on disk) those bunch of packages and folders for the enums, next to enum.go (maybe in appropriately named folder).
2. OR, if golang doesn't easily lend itself to creating a single enum file, you should *generate* as many individual enums files as you need. 

Hope this helps